test/export/note_sorter_spec.py

```diff
@@ -0,0 +1,148 @@
+import logging
+
+from mamba import describe, it, context
+from unittest.mock import MagicMock
+from random import shuffle
+
+from test_utils.anki import mock_anki_modules
+
+mock_anki_modules()
+
+from crowd_anki.export.note_sorter import NoteSorter
+from crowd_anki.config.config_settings import ConfigSettings, NoteSortingMethods
+
+test_guids = ["abc", "bcd", "cde", "def", "efg", "fgh"]
+test_flags = [0, 1, 2, 3, 4, 5]
+test_tags = ["adjectives", "directions", "interesting", "nouns", "verbs", "zzzzFinal"]
+test_notemodels = ["Default", "LL Noun", "LL Sentence", "LL Verb", "LL Word", "Zulu"]
+test_notemodelids = test_guids
+test_fields = test_tags
+
+note_sorting_single_result_pairs = [
+    (NoteSortingMethods.GUID, test_guids),
+    (NoteSortingMethods.FLAG, test_flags),
+    (NoteSortingMethods.TAG, test_tags),
+    (NoteSortingMethods.NOTE_MODEL, test_notemodels),
+    (NoteSortingMethods.NOTE_MODEL_ID, test_notemodelids),
+    (NoteSortingMethods.FIELD1, test_fields),
+    (NoteSortingMethods.FIELD2, test_fields)
+]
+
+test_multikey_notemodel_guid = [(notemodel, guid) for notemodel in test_notemodels for guid in test_guids]
+
+
+class NoteSorterTester:
+    def __init__(self):
+        self.note_sorter = None
+        self.notes = []
+        self.sorted_notes = []
+
+        self.config = ConfigSettings()
+
+    @staticmethod
+    def get_single_note_mock(i):
+        note = MagicMock()
+
+        note.anki_object.guid = test_guids[i]
+        note.anki_object.flags = test_flags[i]
+        note.anki_object.tags = test_tags[i]
+
+        note.anki_object._model = {
+            "name": test_notemodels[i],
+            "crowdanki_uuid": test_notemodelids[i]
+        }
+
+        note.anki_object.fields = [test_fields[i], test_fields[i]]
+
+        return note
+
+    @staticmethod
+    def get_multikey_note_mock(i):
```

Comment:
I'm confused on why multi-key setup is different from single key? you still want the same values set, it's just the order in which you expect them sorted would presumably change

Reply:
I'm not sure if the N^2 setup is required in a first place. You're really testing if the sort method is called and solves your scenario in principle and not if the sort method is correct in itself. So I think testing all possible combinations is redundant for that purpose.

Reply:
Sure, I agree. Though I'd still need a different result at the end to compare it to, and thus would still need a different function to set them to. I agree this part of the class could use some type of tidying up, but there's nothing obvious that jumps out just now :thinking: I suggest we come back to this and leave it like this for now